### PR TITLE
[Console] Throw a TypeError for non-int return value calling Command::execute()

### LINKED ISSUE #33747: Command::execute() should always return int - deprecate returning null
Right now, `Command::execute()` returns `int|void`.

I'd suggest we should deprecate returning void (or null) by throwing a deprecation when the method returns a non-int, in `Application`.

We would need to update all core commands to `return 0;` where they `return;` right now.

And in 5.0, we would make returning a non-int throw an exception, in preparation for adding a real return type in 6.0.

Help wanted.

### PR DESCRIPTION
**Pull request by jschaedl**
| Q             | A
| ------------- | ---
| Branch?       | 5.0
| Bug fix?      | no
| New feature?  | no 
| Deprecations? | yes 
| Tickets       | Fix #33747 
| License       | MIT
| Doc PR        | - 

### Todo

- [x] needs to be rebased after 4.4 was merged into master (see: https://github.com/symfony/symfony/pull/33805)

### CODE REVIEW

File: src/Symfony/Component/Console/Command/Command.php [hunk 1]
```diff
@@ -255,7 +257,7 @@ public function run(InputInterface $input, OutputInterface $output)
             $statusCode = $this->execute($input, $output);
 
             if (!\is_int($statusCode)) {
-                @trigger_error(sprintf('A non numeric or nullable $statusCode returned by Command::execute() is deprecated since Symfony 4.4, return an integer value instead.'), E_USER_DEPRECATED);
+                throw new UnexpectedValueException(sprintf('A non numeric or nullable $statusCode returned by Command::execute() is not allowed anymore since Symfony 5.0, return an integer value instead.'));
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Console/Command/Command.php]: I'd suggest throwing a `TypeError` instead, to prepare for real return types
the message should be future-oriented also.

`throw new \TypeError(sprintf('Return value of "%s::execute()" must be of the type int, %s returned.', \get_class($this), \gettype($statusCode)));`

Comment (jschaedl) [reply src/Symfony/Component/Console/Command/Command.php]: fixed

File: src/Symfony/Component/Console/Command/Command.php [hunk 2]
```diff
@@ -195,7 +195,8 @@ protected function initialize(InputInterface $input, OutputInterface $output)
      *
      * @return int The command exit code
      *
-     * @throws \Exception When binding input fails. Bypass this by calling {@link ignoreValidationErrors()}.
+     * @throws \Exception               When binding input fails. Bypass this by calling {@link ignoreValidationErrors()}.
+     * @throws UnexpectedValueException when Command::execute() returns a non-int value
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Console/Command/Command.php]: this should be removed (no need to mention TypeError)

**Comment by nicolas-grekas** [comment]
Thank you @jschaedl

**Pull request merged**